src/common/FlatList.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+
+// eslint-disable-next-line
+import type { Props as FlatListProps } from 'FlatList';
+
+import * as React from 'react';
+import { FlatList as RNFlatList } from 'react-native';
+import { ON_END_REACHED_THRESHOLD } from '../constants';
+
+type Props<TEntity> = {
+  extraData?: any,
+  innerRef: React.Ref<RNFlatList<TEntity>>,
+} & FlatListProps<TEntity>;
+
+type State = {|
+  isRefreshing: boolean,
+|};
+
+class FlatList<TEntity> extends React.Component<Props<TEntity>, State> {
+  static defaultProps = {
+    ...RNFlatList.defaultProps,
+  };
+
+  state = {
+    isRefreshing: false,
+  };
+
+  _onRefresh = async (): Promise<void> => {
+    if (!this.props.onRefresh) {
+      return;
+    }
+    this.setState(() => ({ isRefreshing: true }));
+    await this.props.onRefresh();
+    this.setState(() => ({ isRefreshing: false }));
+  };
+
+  render() {
+    return (
```

Comment:
nit: You should destructure the props instead of using `this.props`.

I also think `RNFlatList` should take in exact props. Right now you are just using `spread` and overriding a few of the props.  If you destructure it would be a bit cleaner.